crates/fluvio-smartengine/src/smartstream/mod.rs

```diff
@@ -245,7 +248,18 @@ impl dyn SmartStream + '_ {
             let now = Instant::now();
 
             let mut join_record = vec![];
-            join_last_record.encode(&mut join_record, 0)?;
+            if let Some(right_join_stream) = right_join.as_mut() {
+                match right_join_stream.next().await {
```

Comment:
I think that we could have issues here if we dont have new records in the right stream

Reply:
No changes in semantics.  Only difference is that waiting for join is done at batch processing which allow decouple from stream fetch pipeline

Reply:
but my concern here is that I think that this will require a new record per batch,

Reply:
move back join update to inside fetch